src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPageDefault/EnrollmentAddEventPageDefault.container.js

```diff
@@ -33,28 +39,43 @@ export const EnrollmentAddEventPageDefault = ({
 
     const history = useHistory();
     const dispatch = useDispatch();
-    const { fromClientDate } = useTimeZoneConversion();
 
     const handleCancel = useCallback(() => {
         history.push(`enrollment?${buildUrlQueryString({ programId, orgUnitId, teiId, enrollmentId })}`);
     }, [history, programId, orgUnitId, teiId, enrollmentId]);
 
+    const onUpdateEnrollmentStatus = useCallback((enrollmentToUpdate) => {
+        dispatch(updateEnrollmentAndEvents(enrollmentToUpdate));
+    }, [dispatch]);
+
+    const onUpdateEnrollmentStatusError = useCallback((message) => {
+        dispatch(rollbackEnrollmentAndEvents());
+        dispatch(showEnrollmentError({ message }));
+    }, [dispatch]);
+
+    const onUpdateEnrollmentStatusSuccess = useCallback(({ redirect }) => {
+        dispatch(removeExternalEnrollmentStatus());
+        dispatch(commitEnrollmentAndEvents());
+        redirect && history.push(`enrollment?${buildUrlQueryString({ programId, orgUnitId, teiId, enrollmentId })}`);
```

Comment:
Hey @JoakimSM, yes, that is correct. The request is aborted if I navigate away from the page in `onUpdateEnrollmentStatus` and I couldn't figure out how to fix it. Do you have any suggestions? 🤔 

PS: I think in the long run it can be solved by using react-query instead of the app-runtime hooks. However, I chose to postpone refactoring the enrollment widget to use react-query because I think it needs a proper tech design and is outside the scope of this feature.

Reply:
I agree with your conclusion. Let's leave it like this for now.